packages/dds/sequence/src/test/revertibles.perf.spec.ts

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BenchmarkType, benchmark } from "@fluid-tools/benchmark";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils";
+import { SharedString } from "../sharedString";
+import { SharedStringFactory } from "../sequenceFactory";
+import { IIntervalCollection, IntervalType, SequenceInterval } from "../intervalCollection";
+import {
+	appendChangeIntervalToRevertibles,
+	appendSharedStringDeltaToRevertibles,
+	SharedStringRevertible,
+} from "../revertibles";
+
+describe("Interval revertible perf", () => {
+	describe("Move interval", () => {
+		let collection: IIntervalCollection<SequenceInterval>;
+		let interval: SequenceInterval;
+	
+		const setUp = (enableRevertibles: boolean) => {
+			const runtime = new MockFluidDataStoreRuntime();
+			runtime.options = { mergeTreeUseNewLengthCalculations: true };
+			const factory = new SharedStringFactory();
+			const sharedString = factory.create(runtime, "id");
+			sharedString.insertText(0, "a".repeat(100));
+			collection = sharedString.getIntervalCollection("test");
+			interval = collection.add(45, 77, IntervalType.SlideOnRemove);
+			if (enableRevertibles) {
+				collection.on("changeInterval", (i, previousInterval, local, op) => {
+					const revertibles: SharedStringRevertible[] = [];
+					appendChangeIntervalToRevertibles(sharedString, i, previousInterval, revertibles);
+				});
+			}
+		};
+
+		const execute = () => {
+			collection.change(interval.getIntervalId(), 1, 2);
+		};
+
+		benchmark({
+			beforeEachBatch: () => {
+				setUp(false);
+			},
+			type: BenchmarkType.Perspective,
+			title: "without revertible",
+			benchmarkFn: () => {
+				execute();
+			},
+		});
+
+		benchmark({
+			beforeEachBatch: () => {
+				setUp(true);
+			},
+			type: BenchmarkType.Measurement,
+			title: "with revertible",
+			benchmarkFn: () => {
+				execute();
+			},
+		});
+	});
+
+	describe("Add and remove range with interval and moved interval", () => {
+		let sharedString: SharedString;
+		let collection: IIntervalCollection<SequenceInterval>;
+		let interval: SequenceInterval;
+
+		const setUp = (enableRevertibles: boolean) => {
+			const runtime = new MockFluidDataStoreRuntime();
+			runtime.options = { mergeTreeUseNewLengthCalculations: true };
+			const factory = new SharedStringFactory();
+			sharedString = factory.create(runtime, "id");
+			sharedString.insertText(0, "abc");
```

Comment:
I'm not sure how long the benchmark tests run, but could you just create a string with "abc" repeated 100 or 1000 times for this test here instead of adding text in the execute function?

Reply:
might be a reasonable use case for `benchmarkFnCustom`, which is basically an escape hatch for problems like this.

Reply:
ok, yeah after reading I think you probably want to use `benchmarkFnCustom` to make sure it's a fair comparison. `Insert ${nodeCount} nodes, ${identifierDensityPercentage}% of which have identifiers` in `nodeIdentifierIndex.bench.ts` is an example usage of a test case with a pretty similar problem. You'll likely want to fix the number of insertions to some reasonable constant (e.g. 100) and follow a similar pattern to that test case.